device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/IotHubSasTokenAuthenticationProvider.java

```diff
@@ -120,7 +120,7 @@ public boolean shouldRefreshToken(boolean proactivelyRenew)
 
     public int getMillisecondsBeforeProactiveRenewal()
     {
-        return (int) this.tokenValidSecs * this.timeBufferPercentage;
+        return (int) ((this.tokenValidSecs * 1000) * (this.timeBufferPercentage / 100.0));
```

Comment:
What is divide by 100 for?

Reply:
The field ```timeBufferPercentage``` is supposed to represent 85%, but it is just 85 instead of .85

Reply:
Worth a code comment.